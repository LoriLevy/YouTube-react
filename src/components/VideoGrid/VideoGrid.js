import React from 'react';
import './VideoGrid.scss';
import { Divider } from 'semantic-ui-react';
import VideoPreview from '../VideoPreview/VideoPreview';
import { VideoGridHeader } from './VideoGridHeader/VideoGridHeader';

function VideoGrid(props) {
  const divider = props.hideDivider ? null : <Divider />;
  return (
    <>
      <VideoGridHeader title={props.title} />

      <div className='video-grid'>
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
      </div>
      {divider}
    </>
  );
}
export default VideoGrid;
