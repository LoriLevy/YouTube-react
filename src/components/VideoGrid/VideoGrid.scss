.video-grid {
  display: flex; /* align elements horizontally */
  flex-wrap: wrap; /* just in case there is not enough room */

  & > * {
    margin-left: 4px;
    margin-bottom: 24px;
  }
}
