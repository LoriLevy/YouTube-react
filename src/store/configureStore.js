import {applyMiddleware, createStore, compose} from 'redux';
import reducer from './reducers';
import createSagaMiddleware from 'redux-saga';
import rootSaga from './sagas';

export function configureStore() {
    const sagaMiddleware = createSagaMiddleware();
    //added composeEnhancers for seeing history of actions
    const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
    const store = createStore(reducer, 
        composeEnhancers(applyMiddleware(sagaMiddleware))
    );

    sagaMiddleware.run(rootSaga);
    return store;
}

/* returns __REDUX_DEVTOOLS_EXTENSION_COMPOSE__ */