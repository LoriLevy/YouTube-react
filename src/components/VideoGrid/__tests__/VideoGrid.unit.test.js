import React from "react";
import { shallow } from "enzyme";
import VideoGrid from "../VideoGrid";
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import {  Provider  } from 'react-redux';

const middleware = [thunk];
const mockStore = configureStore(middleware);
const initialState = {
  example: {}
};
const store = mockStore(initialState);

describe("VideoGrid", () => {
  test("renders without props", () => {
    const wrapper = shallow(<Provider store={store}><VideoGrid /></Provider>);
    expect(wrapper).toMatchSnapshot();
  });

  test("renders with a title prop", () => {
    const wrapper = shallow(<Provider store={store}><VideoGrid title="Trending" /></Provider>);
    expect(wrapper).toMatchSnapshot();
  });

  test("renders without a divider", () => {
    const wrapper = shallow(<Provider store={store}><VideoGrid hideDivider={true} /></Provider>);
    expect(wrapper).toMatchSnapshot();
  });
});
