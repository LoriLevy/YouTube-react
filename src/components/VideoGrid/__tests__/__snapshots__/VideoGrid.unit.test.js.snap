// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoGrid renders with a title prop 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <VideoGrid
    title="Trending"
  />
</ContextProvider>
`;

exports[`VideoGrid renders without a divider 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <VideoGrid
    hideDivider={true}
  />
</ContextProvider>
`;

exports[`VideoGrid renders without props 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <VideoGrid />
</ContextProvider>
`;
